### DISCUSSION TITLE
Retrofit 2.9.0 on API levels lower than 21

### DISCUSSION THREAD

**Discussion by jokoolle-183** [square/retrofit#3525] [Q&A]
Hi guys,

We have an SDK which uses the latest version of the Retrofit library- 2.9.0. The SDK supports minimum API level 16, and Retrofit requires minimum API level 21, which causes the SDK to fail to initialize.

According to this [blog post](https://cashapp.github.io/2019-02-05/okhttp-3-13-requires-android-5), attached to Retrofit 2.7.0 release changes, if we want support for **_4.x_** Android devices, we would need to use the **_3.12.x_** version of OkHttp library. Is there a way to set this version of OkHttp within the Retrofit library? If not, is there a workaround to make the Retrofit library work on API levels lower that 21?

**Comment by ivasku** [comment]
I would like to know this too, @jokoolle-183 did you maybe found the information about the Min Api level ? How to use it for example for Api 19 ?

**Comment by hellbill** [comment]
https://github.com/JakeWharton/timber/releases/tag/5.0.1

**Comment by Poonam4789** [comment]
While uploading file using multipart using retrofit 2.9.0,it convert file to binary and adds mluc tag along with that , because of this tag api is blocking at our server because it has firewall and and does not allows mluc tags , can anyone help me resolve this issue .
why this tag is added by retrofit ?

**Logs:**
upload_document
--21e712b2-520c-4844-9cb8-219d9a77d1cc
Content-Disposition: form-data; name="file"; filename="1644842874898_croppedImg.jpg"
Content-Type: multipart/form-data
Content-Length: 45195

......JFIF.............(ICC_PROFILE...............mntrRGB XYZ ............acsp.......................................-................................................... desc.......trXYZ...d....gXYZ...x....bXYZ........rTRC.......(gTRC.......(bTRC.......(wtpt........cprt.......< mluc ............enUS...X.....s.R.G.B................................................................................XYZ ......o...8.....XYZ ......b.........XYZ ......$.........para..........ff......
Y......
[........XYZ ............... -mluc ............enUS... .....G.o.o.g.l.e. .I.n.c... .2.0.1.6...C...............
.

...............%...#... , #&')*)..-0-(0%()(...C....
.
